### PR TITLE
NMS-8024: Add support for OSGi-based Ticketing plugins.

### PR DESCRIPTION
**Pull request by j-white**
* Moved ticketd out of opennms-services and into it's own module
* Added a new OSGiBasedTicketerPlugin that uses a plugin from the OSGi service registry
* Updated the JIRA ticketing plugin to use the latest client API bindings
* Moved the JIRA ticket plugin to use the new OSGi interface
* Documented how to setup the JIRA plugin

JIRA: http://issues.opennms.org/browse/NMS-8024
Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS325

### CODE REVIEW

File: features/ticketing/daemon/src/main/java/org/opennms/netmgt/ticketd/DefaultTicketerServiceLayer.java [hunk 1]
```diff
@@ -114,16 +102,16 @@ public void afterPropertiesSet() throws Exception {
     public void cancelTicketForAlarm(int alarmId, String ticketId) {
         OnmsAlarm alarm = m_alarmDao.get(alarmId);
         if (alarm == null) {
-            throw new ObjectRetrievalFailureException("Unable to locate Alarm with ID: "+alarmId, null);
+            LOG.error("No alarm with id {} was found. Ticket with id '{}' will not be canceled.", alarmId, ticketId);
```
Comment (soleger) [inline features/ticketing/daemon/src/main/java/org/opennms/netmgt/ticketd/DefaultTicketerServiceLayer.java]: Oh no, it's the Canadian spelling of 'canceled'!

File: dependencies/pom.xml [hunk 2]
```diff
@@ -28,7 +28,6 @@
     <module>javamail</module>
     <module>jaxb</module>
     <module>jcifs</module>
-    <module>jersey</module>
     <module>jersey-client</module>
```
Comment (soleger) [inline dependencies/pom.xml]: With these changes, the only things left using jersey-client-dependencies appear to be features/ticketing and features/resource-cli. Does that sound correct?

Comment (j-white) [inline dependencies/pom.xml]: features/reporting/repository still relies on it too. We could work on migrating these to use cxf though.

File: dependencies/jersey/pom.xml [hunk 3]
```diff
@@ -1,155 +0,0 @@
...
-  <repositories>
-    <repository>
-      <snapshots><enabled>false</enabled></snapshots>
-      <releases><enabled>true</enabled></releases>
-      <id>maven2-repository.dev.java.net</id>
-      <name>Java.net Repository for Maven</name>
-      <url>http://maven.opennms.org/content/repositories/maven2-repository.dev.java.net/</url>
-    </repository>
-  </repositories>
-</project>
```
Comment (mvrueden) [inline dependencies/jersey/pom.xml]: R.I.P.

File: features/ticketing/api/pom.xml [hunk 4]
```diff
@@ -8,6 +8,21 @@
...
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.felix</groupId>
+        <artifactId>maven-bundle-plugin</artifactId>
+        <extensions>true</extensions>
+        <configuration>
+          <instructions>
+            <Bundle-RequiredExecutionEnvironment>JavaSE-1.8</Bundle-RequiredExecutionEnvironment>
```
Comment (mvrueden) [inline features/ticketing/api/pom.xml]: This is not really part of this pull request, but I saw this now more often.
We have properties for everything, except this. How about having a JdkVersion8String and a JdkVersion7String or something like that defined in the root pom. Would make migration easier in the future.

File: features/ticketing/jira-integration/pom.xml [hunk 5]
```diff
@@ -7,55 +7,36 @@
...
+              <Bundle-RequiredExecutionEnvironment>JavaSE-1.8</Bundle-RequiredExecutionEnvironment>
+            </instructions>
+          </configuration>
+        </plugin>
+        <plugin>
+            <groupId>org.opennms.maven.plugins</groupId>
+            <artifactId>features-maven-plugin</artifactId>
+            <configuration>
+                <bundles>
+                    <bundle>mvn:joda-time/joda-time/${jodaTimeVersion}</bundle>
```
Comment (mvrueden) [inline features/ticketing/jira-integration/pom.xml]: You do not have to define the bundles manually if they are defined in the dependencies as not provided and are real bundles.

File: features/ticketing/api/pom.xml (hunk 4 above)
Comment (soleger) [inline features/ticketing/api/pom.xml]: I agree. We should call it osgiExecutionEnvironment just to make its usage explicit. Then we can change it later on to Java-1.9 and the variable name doesn't change.

Comment (mvrueden) [inline features/ticketing/api/pom.xml]: We also need another one, maybe `osgiExecutionEnvironmentLegacy` for the previous version of the JRE. We have at least one bundle which requires JRE 7 and not 8.

**Pull request merged**